freckle-app/tests/Freckle/App/Test/Hspec/AnnotatedExceptionSpec.hs

```diff
@@ -0,0 +1,147 @@
+module Freckle.App.Test.Hspec.AnnotatedExceptionSpec
+  ( spec
+  ) where
+
+import Freckle.App.Prelude
+
+import Data.Annotation (toAnnotation)
+import Freckle.App.Exception (AnnotatedException (..))
+import Freckle.App.Test.Hspec.AnnotatedException (annotateHUnitFailure)
+import GHC.Exts (fromList)
+import GHC.Stack (CallStack, SrcLoc (..))
+import Test.HUnit.Lang (FailureReason (..), HUnitFailure (..))
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+spec :: Spec
+spec = do
+  describe "annotateHUnitFailure" $ do
+    describe "does nothing if there are no annotations" $ do
+      it "when the failure is Reason" $
+        let e = HUnitFailure Nothing (Reason "x")
+        in  annotateHUnitFailure (AnnotatedException [] e) `shouldBe` e
+
+      it "when the failure is ExpectedButGot with no message" $
+        let e = HUnitFailure Nothing (ExpectedButGot Nothing "a" "b")
+        in  annotateHUnitFailure (AnnotatedException [] e) `shouldBe` e
+
+      it "when the failure is ExpectedButGot with a message" $
+        let e = HUnitFailure Nothing (ExpectedButGot (Just "x") "a" "b")
+        in  annotateHUnitFailure (AnnotatedException [] e) `shouldBe` e
+
+    describe "can show an annotation" $ do
+      it "when the failure is Reason" $
+        annotateHUnitFailure
+          AnnotatedException
+            { annotations = [toAnnotation @Int 56]
+            , exception = HUnitFailure Nothing (Reason "x")
+            }
+          `shouldBe` HUnitFailure
+            Nothing
+            ( Reason
+                "x\n\
+                \\n\
+                \Annotations:\n\
+                \\t * Annotation @Int 56"
```

Comment:
FWIW, I find the literal newlines and continuation escapes make this hard to scan. Howe about something like,

```hs
Reason $ unlines
  [ "x"
  , ""
  , "Annotations:"
  , "    t * Annotation @Int 56"
  ]
```

Does it need to be easy to scan? Meh, not really. But I think a bug in the implementation string building, that was just offset by a bug in _this_ string building, would be hard to spot. I also just like, when testing string building code, to actually be able to understand the built string in test.